### PR TITLE
Setup associations

### PR DESCRIPTION
**Pull request by zbahati**
# Ft setup association

Add classes book, rental, classroom
Add relationship rentals to the person
Add relationship book, to the student
Add relationship of book to the rental and parent

### CODE REVIEW

File: classroom_class.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :lable
+
```
Comment (TadesseDev) [inline classroom_class.rb]: - [ ] Good job adding the `attr_accessor` method @zbahati, but alongside that, we need to have `attr_reader` for the `students` attribute. Please consider adding this one.

File: student_class.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require './person_class'
+
+class Student < Person
+  attr_accessor :classroom
+
```
Comment (TadesseDev) [inline student_class.rb:5]: - [ ] Dear @zbahati, for students we need to have a method that serves as the 'belongs-to' side. Hence we need to create a method that takes a classroom and adds the student to the classroom unless the classroom already has that student. 
 **I see you put this logic inside the constructor, but there might be a chance the student is created without a classroom at first.**
> Something like 
```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: book_class.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Book
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
+end
```
Comment (TadesseDev) [inline book_class.rb]: - [ ] Also, for the books, we need to keep a list of `rentals` and add every rental that is made to its rental list. Please create a rental array and add it to the `attr_accessor` list, besides that, create a `method` to add a new rental to the rentals list. 

**I notice you implement this feature with the `Person` class, by adding rental to the book while adding a person, this approach is highly coupled and I would strongly advise not to follow it. The `book` class should solely be responsible for maintaining the association between the `book and rental`.**

File: person_class.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  def correct_name
+    @name
+  end
+
+  def add_rental(book)
+    rental = Rental.new(data, book, self)
+    book.add_rental(rental)
+  end
```
Comment (TadesseDev) [inline person_class.rb]: - [Optional] Dear @zbahati, while creating a rental, consider a situation where the `date` is not the current date. Imagine the book is borrowed a day before, but it's being registered today. I would suggest accepting the `date` attribute as a parameter and creating the rental with it.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @zbahati 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] All the class are created in separate files. ✅
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @zbahati,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- No linter errors ✔️
- Correct workflow was used ✔️
- Well organized and clean code ✔️

# Status: Approved ✔️✔️✔️

approv


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**